### PR TITLE
fix: prevent temporary update of sub-path value in store object 

### PR DESCRIPTION
**Pull request by ohansFavour**
## Description

Previous storeValue implementation has below code to temporarily store the value to dataTree while running evaluation.

// momentarily store this value in local state to support loops
   ` _.set(self, `appsmith.store[${key}]`, value);`
Here, as we are using lodash's set, keys like test and test.two are not treated as two different keys and hence the bug.

New implementation

`_.set(self, ["appsmith", "store", key], value);`

Fixes https://github.com/appsmithorg/appsmith/issues/14827

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Cypress

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Rishabh-Rathod — DISMISSED** [review]
Code Changes looks good to me.

File: app/client/src/sagas/ActionExecution/StoreActionSaga.ts [hunk 1]
```diff
@@ -22,7 +23,7 @@ export default function* storeValueLocally(
     const appStoreName = getAppStoreName(applicationId, branch);
     const existingStore = localStorage.getItem(appStoreName) || "{}";
     const parsedStore = JSON.parse(existingStore);
-    parsedStore[action.key] = action.value;
+    _.set(parsedStore, action.key, action.value);
```
Comment (Rishabh-Rathod) [inline app/client/src/sagas/ActionExecution/StoreActionSaga.ts:26]: This doesn’t seem to be an expected change, was this added by mistake while adding previous PR's code?

**Review by Rishabh-Rathod — COMMENTED** [review]
@ohansFavour  could you please check this once.

File: app/client/src/sagas/ActionExecution/StoreActionSaga.ts (hunk 1 above)
Comment (Aishwarya-U-R) [reply app/client/src/sagas/ActionExecution/StoreActionSaga.ts]: Oh, I'm sorry... I was trying to pull latest of this branch & seems to have messed it..

File: app/client/src/workers/Actions.ts [hunk 2]
```diff
@@ -77,7 +77,7 @@ const DATA_TREE_FUNCTIONS: Record<
   },
   storeValue: function(key: string, value: string, persist = true) {
     // momentarily store this value in local state to support loops
-    _.set(self, ["appsmith", "store", key], value);
+    _.set(self, `appsmith.store[${key}]`, value);
```
Comment (Rishabh-Rathod) [inline app/client/src/workers/Actions.ts]: I have not tried using square brackets for `set` before. 
Does that also works for the similar to `_.set(self, ["appsmith", "store", key], value);` ?

Comment (ohansFavour) [reply app/client/src/workers/Actions.ts]: `_.set(self, `appsmith.store[${key}]`, value)` will update the path.

`_.set(self, ["appsmith", "store", key], value)` won't.

Kindly confirm that you are reviewing the most recent change.

Comment (Rishabh-Rathod) [reply app/client/src/workers/Actions.ts]: Changes which i was looking at weren't latest hence the confusion. Thanks for resolving it.

**Review by Aishwarya-U-R — APPROVED** [review]
Approving cypress changes!

**Pull request merged**